Energy Saver/Model/Serialization.cs

```diff
@@ -28,11 +28,24 @@ public static List<List<Taxes>> ReadFromFile(this PageModel pageModel)
         return taxes;
     }
 
-    public static void WriteToFile(this PageModel pageModel, Taxes taxes)
+    public static void WriteText(this PageModel pageModel, List<Taxes> taxes)
     {
-        List<Taxes>? temp = ReadText();
+        string serializedString = JsonConvert.SerializeObject(temp, Formatting.Indented, GetSerializerSettings(new CamelCasePropertyNamesContractResolver()));
+
+        File.WriteAllText(Path.GetFullPath(path), serializedString);
+    }
+
+    public static void WriteEntryToFile(this PageModel pageModel, Taxes taxes)
+    {
+        List<Taxes>? newList = ReadText();
+
+        int duplicateIndex = newList.FindIndex(tax => tax.Year.Equals(taxes.Year) && tax.Month.Equals(taxes.Month));
 
-        temp.Add(taxes);
+        //If duplicate was found
+        if (duplicateIndex != -1)
+            newList[duplicateIndex] = taxes;
+        else
+            newList.Add(taxes);
 
         string serializedString = JsonConvert.SerializeObject(temp, Formatting.Indented, GetSerializerSettings(new CamelCasePropertyNamesContractResolver()));
```

Comment:
There is some code duplication. 2 lines are the same as the WriteText method

Reply:
couldn't find a better solution, will look into it later